apps/src/blockly/addons/cdoFieldDropdown.ts

```diff
@@ -194,6 +194,37 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     (this as any).arrow = arrow;
   }
+
+  /**
+   * Get the text from this field to display on the block. May differ from
+   * `getText` due to ellipsis, and other formatting.
+   * @override Handling of text for RTL blocks is customized.
+   * See: https://github.com/google/blockly/issues/8645
+   * @returns Text to display.
+   */
+  protected getDisplayText_(): string {
+    let text = this.getText();
+    if (!text) {
+      // Prevent the field from disappearing if empty.
+      return GoogleBlockly.Field.NBSP;
+    }
+    if (text.length > this.maxDisplayLength) {
+      // Truncate displayed string and add an ellipsis ('...').
+      text = text.substring(0, this.maxDisplayLength - 2) + '…';
```

Comment:
Will we use the proper substring here if we overflow the display _and_ are in RTL? We're not chopping off the wrong end of the string, are we?

Reply:
This apparently needs to merge today due to the HoC launch. Since I don't believe we have any/many long string options, I'm going to take a closer look at this as a follow-up.
FWIW, Rachel indicated they'd also except this change up stream, with another question to poke at: https://github.com/google/blockly/issues/8645#issuecomment-2467674211